_rules/aria-role-permitted-j7zzqr.md

```diff
@@ -0,0 +1,253 @@
+---
+id: j7zzqr
+name: ARIA role is permitted for the element.
+rule_type: atomic
+description: |
+  This rule checks that explicit WAI-ARIA roles are allowed for the element they are specified on.
+accessibility_requirements:
+  html-aria:docconformance:
+    title: ARIA in HTML, 4. Document conformance requirements for use of ARIA attributes in HTML
+    forConformance: true
+    failed: not satisfied
+    passed: satisfied
+    inapplicable: satisfied
+input_aspects:
+  - Accessibility Tree
+  - DOM Tree
+acknowledgments:
+  authors:
+    - Jean-Yves Moyen
+---
+
+## Applicability
+
+This rule applies to any [HTML element][namespaced element] which has an [explicit semantic role][explicit role] and is not [programmatically hidden][].
+
+## Expectation
+
+For each test target, its [explicit semantic role][explicit role] is allowed on this element, according to the [ARIA in HTML specification][aria in html document conformance].
+
+## Assumptions
+
+There are no assumptions.
+
+## Accessibility Support
+
+There are no accessibility support issues known.
+
+## Background
+
+Each element also has an [implicit semantic role][implicit role] defined in the [HTML Accessibility API Mappings][html aam:roles]. Setting the [explicit role][] as the same as the [implicit role][] is not recommended but nonetheless allowed. This rule does not check that scenario.
+
+This rule does not check specifically whether the [explicit role][] is deprecated, or whether it is the special `generic` role that should not be used by authors. These are not conformance requirement, or are checked by other rules.
+
+This rule only check for elements that are not [programmatically hidden][]. Elements that are [programmatically hidden][], e.g. due to a CSS `display:none`, must still have valid roles in order to conform to [ARIA in HTML][aria in html document conformance]. However, as long as the elements with the errant roles remain programmatically hidden, they are not expected to cause accessibility issues. This rule only checks a snapshot of a page in a given state; if interacting with the page programmatically exposes an element with an errant role that this rule previously ignored, the new state of the page should be reevaluated to ensure it still passes this rule, and that an accessibility issue has not been introduced to users. Examples only show [programmatically hidden][] elements with a correct role.
```

Comment:
just a missing 's' i had previously not noticed in the first sentence.  otherwise lgtm.

```suggestion
This rule only checks for elements that are not [programmatically hidden][]. Elements that are [programmatically hidden][], e.g. due to a CSS `display:none`, must still have valid roles in order to conform to [ARIA in HTML][aria in html document conformance]. However, as long as the elements with the errant roles remain programmatically hidden, they are not expected to cause accessibility issues. This rule only checks a snapshot of a page in a given state; if interacting with the page programmatically exposes an element with an errant role that this rule previously ignored, the new state of the page should be reevaluated to ensure it still passes this rule, and that an accessibility issue has not been introduced to users. Examples only show [programmatically hidden][] elements with a correct role.
```